packages/backend/src/plugins/badges.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createRouter } from '@backstage/plugin-badges-backend';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin({
+  logger,
+  config,
+  discovery,
+}: PluginEnvironment) {
+  return await createRouter({ logger, config, discovery });
```

Comment:
What do you think about passing in the list of badges here instead of through static config? I'm thinking badges aren't really going to vary per deployment, and it bloats the configuration quite a bit on top of having to implement a custom templating language for the values. If we make it configured through code here you can rely on callbacks that receive the entity and whatever else might be needed, on top of having access to config and custom logic.

Reply:
Sounds good. Still think there would be some form of templating going on, to get dynamic values in there, somehow though. (not too much custom templating though, leaving the heavylifting to js template literals ;) )

Makes sense also to define the badges in code, as you say, they'll probably pretty much alike in most deployments.

Reply:
Ok, I've made a semi-refactoring regarding how and where the badges are defined.

So, there are no callbacks and custom logic, but I do declare the badges in code, and load them along when creating the backend router. I've kept the config source as an option (easy to add one off custom badges etc..) as it was already there..

Also, the frontend now relies exclusively on backend api:s to fetch all required badge information, so does no config introspection of its own.